pkg/nsx/services/subnet/builder.go

```diff
@@ -0,0 +1,99 @@
+package subnet
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+var (
+	String = common.String
+	Int64  = common.Int64
+	Bool   = common.Bool
+)
+
+func getCluster(service *SubnetService) string {
+	return service.NSXConfig.Cluster
+}
+
+func (service *SubnetService) buildSubnet(obj client.Object, tags []model.Tag) (*model.VpcSubnet, error) {
+	tags = append(tags, service.buildBasicTags(obj)...)
+	var nsxSubnet *model.VpcSubnet
+	switch o := obj.(type) {
+	case *v1alpha1.Subnet:
+		nsxSubnet = &model.VpcSubnet{
+			Id:         String(string(o.GetUID())),
```

Comment:
For now, the NSX subnet's name is uid because we didn't explicitly set the display name. I think it'd better to constructed by name instead of uid.

Reply:
ok, done.